### PR TITLE
Utils tests

### PR DESCRIPTION
**Pull request by amrendranath**

### CODE REVIEW

File: facade-app/src/test/Utils.test.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import {
```
Comment (preetkaran20) [inline facade-app/src/test/Utils.test.tsx:1]: the build is failing, can you please fix that.

Comment (amrendranath) [reply facade-app/src/test/Utils.test.tsx:1]: Working on it.

Comment (amrendranath) [reply facade-app/src/test/Utils.test.tsx:1]: @preetkaran20 I am not sure what's the issue. Can anyone help me with this?

Comment (preetkaran20) [reply facade-app/src/test/Utils.test.tsx:1]: Error is:
```
FAIL src/test/Utils.test.tsx (10.173 s)
  ● getResource() › should call the callback function with the obj when isJson is true

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      20 |
      21 | describe("getResource()", () => {
    > 22 |   it("should call the callback function with the obj when isJson is true", (done) => {
         |   ^
      23 |     //@ts-ignore
      24 |     const fetchMock = jest.spyOn(global, "fetch").mockResolvedValueOnce({
      25 |       ok: true,

      at src/test/Utils.test.tsx:22:3
      at Object.<anonymous> (src/test/Utils.test.tsx:21:1)

  ● getResource() › should call the callback function with the text when isJson is true

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      43 |   });
      44 |
    > 45 |   it("should call the callback function with the text when isJson is true", (done) => {
         |   ^
      46 |     //@ts-ignore
      47 |     const fetchMock = jest.spyOn(global, "fetch").mockRejectedValueOnce({
      48 |       ok: false,

      at src/test/Utils.test.tsx:45:3
      at Object.<anonymous> (src/test/Utils.test.tsx:21:1)
```

File: facade-app/src/test/Utils.test.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+interface CallbackParams {
+  isSuccessful: boolean;
+  data?: any;
+  error?: any;
+}
+
+const mockData = { res: [] };
+
+// This test is not working in this jest version. But it is working on latest version. This is a bug.
```
Comment (preetkaran20) [inline facade-app/src/test/Utils.test.tsx:17]: may be it is related to this one.

File: facade-app/src/test/Utils.test.tsx (hunk 1 above)
Comment (preetkaran20) [reply facade-app/src/test/Utils.test.tsx:1]: is this test working in your local? can you please compare versions which you are using locally. The github action which is running is present at: https://github.com/SasanLabs/VulnerableApp-facade/blob/main/.github/workflows/Compile.yml

File: facade-app/package-lock.json [hunk 3]
```diff
@@ -17864,9 +17875,11 @@
       }
     },
     "node_modules/type-fest": {
-      "version": "0.21.3",
-      "resolved": "https://registry.npmjs.org/type-fest/-/type-fest-0.21.3.tgz",
-      "integrity": "sha512-t0rzBq87m3fVcduHDUFhKmyyX+9eo6WQjZvf51Ea/M0Q7+T374Jp1aUiyUl0GKxp8M/OETVHSDvmkyPgvX+X2w==",
+      "version": "0.13.1",
```
[the remark this answers is not in the document]
Comment (amrendranath) [reply facade-app/package-lock.json:17878]: Shall I try upgrading the package?

[the remark this answers is not in the document]
Comment (preetkaran20) [reply facade-app/package-lock.json:17878]: we can update jest version

[the remark this answers is not in the document]
Comment (amrendranath) [reply facade-app/package-lock.json:17878]: I don't think its a good idea to update it. See this https://stackoverflow.com/a/39932457/9316358

[the remark this answers is not in the document]
Comment (preetkaran20) [reply facade-app/package-lock.json:17878]: ok then can you please check the work arounds mentioned in the github issue and see if those can work ?

[the remark this answers is not in the document]
Comment (amrendranath) [reply facade-app/package-lock.json:17878]: Those workarounds are also not working for me.

[the remark this answers is not in the document]
Comment (amrendranath) [reply facade-app/package-lock.json:17878]: Let me do some more research on that.

[the remark this answers is not in the document]
Comment (preetkaran20) [reply facade-app/package-lock.json:17878]: hmm, ok try for some more time and if still it doesn't work then let's ignore these 2 tests and we will merge with other test cases.

File: facade-app/package-lock.json [hunk 4]
```diff
@@ -3728,6 +3728,17 @@
         "url": "https://github.com/sponsors/sindresorhus"
       }
     },
+    "node_modules/ansi-escapes/node_modules/type-fest": {
```
Comment (preetkaran20) [inline facade-app/package-lock.json:3731]: can you please remove this change?

File: facade-app/src/test/Utils.test.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
+import {
+  // getResource,
+  removeChildByParentElementId,
+  appendStaticResourcesToDocument,
+  manipulateDOM,
+} from "../Utilities/Utils";
+import { ResourceType } from "../interface/State";
+
+// interface CallbackParams {
```
Comment (preetkaran20) [inline facade-app/src/test/Utils.test.tsx:9]: I think we can remove these commented tests.

Comment (amrendranath) [reply facade-app/src/test/Utils.test.tsx:9]: I will remove it.

**Review by preetkaran20 — APPROVED** [review]

**Comment by preetkaran20** [comment]
LGTM, Thank you !!!.

**Pull request merged**